src/app/shared/table/table-data/table-data.component.ts

```diff
@@ -46,17 +46,17 @@ export class TableDataComponent implements OnChanges {
     return this.config.format?.type;
   }
 
-  translateData(rawData: string): string {
+  translateData(rawData: string | string[]): string {
     if (rawData === undefined || rawData === null) {
       return rawData;
     }
-    let translatedData: string = String(rawData);
-    if (this.config.translationName) {
-      // @ts-ignore
-      translatedData = enums[this.config.translationName][rawData];
-      translatedData = this.translateService.instant(translatedData);
-    }
-    return translatedData;
+    const data = rawData instanceof Array ? [...rawData] : [rawData];
+    return data
+      .map((item) =>
+        // @ts-ignore
+        this.translateService.instant(enums[this.config.translationName][item] || ' ')
```

Comment:
if no translation is found, shouldn't we have a default one? or empty space will do? ... also, if it remains empty space, please go with empty string '' ; not ' '

Reply:
this is just a guard in case enum[key] returns undefined -> it would throw an error in the console
showing nothing should be the norm in this case